### PR TITLE
Reimplement parallelization API

### PR DESCRIPTION
**Pull request by InnocentBug**
This reimplements @johnaparker's parallel API (supersedes #104)

### CODE REVIEW

**Comment by pabloferz** [comment]
Fixes #54

**Comment by InnocentBug** [comment]
I successfully ran OpenMM and Hoomd-blue examples in berlin.
If the test pass, we can go ahead with this.

File: .gitignore [hunk 1]
```diff
@@ -6,4 +6,11 @@ docs/build
...
-examples/*/*gsd
\ No newline at end of file
+examples/*/*gsd
+examples/*/*/*pdf
+examples/*/*/*dat
+examples/*/*/*gsd
+dy_periodic_sirens_sobolev_fit.pdf
+examples/*/.ipynb_checkpoints/
+y_mlp_fit.pdf
+y_periodic_sirens_sobolev_fit.pdf
```
Comment (pabloferz) [inline .gitignore:16]: I would use `*.pdf` instead.

Comment (InnocentBug) [reply .gitignore]: sure, NP

**Comment by pabloferz** [comment]
I can't figure out why the hoomd example is not working here. I wonder if we should remove that form our CI for now and merge and figure out the issue later.

**Review by pabloferz — DISMISSED** [review]

**Review by pabloferz — APPROVED** [review]

**Pull request merged**